### PR TITLE
Fixes 5% fee when contribution fails

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/5594

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- use staging
- enable rewards
- connect KYC account
- add some funds
- tip verified publisher
- log in into uphold sandbox
- make sure that you see 5% amount from your tip in `5% fee` card

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc [hunk 1]
```diff
@@ -76,12 +66,29 @@ void Uphold::StartContribution(const std::string &viewing_id,
...
+void Uphold::ContributionCompleted(
+    ledger::Result result,
+    bool created,
+    const std::string& viewing_id,
+    const double fee,
+    const ledger::ExternalWallet& wallet) {
   const auto reconcile = ledger_->GetReconcileById(viewing_id);
   const auto amount = ConvertToProbi(std::to_string(reconcile.fee_));
 
+  if (result == ledger::Result::LEDGER_OK) {
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc]: we needed to move 5% fee transaction after publisher transaction was successful. Before both were triggered at the same time

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc [hunk 2]
```diff
@@ -76,12 +66,29 @@ void Uphold::StartContribution(const std::string &viewing_id,
   }
 }
 
-void Uphold::ContributionCompleted(ledger::Result result,
-                                   bool created,
-                                   const std::string &viewing_id) {
+void Uphold::ContributionCompleted(
+    ledger::Result result,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc]: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc [hunk 3]
```diff
@@ -76,12 +66,29 @@ void Uphold::StartContribution(const std::string &viewing_id,
...
+    bool created,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc]: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.h [hunk 4]
```diff
@@ -71,9 +71,12 @@ class Uphold {
     GetUserCallback callback);
 
  private:
-  void ContributionCompleted(ledger::Result result,
-                             bool created,
-                             const std::string &viewing_id);
+  void ContributionCompleted(
+      ledger::Result result,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.h]: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.h [hunk 5]
```diff
@@ -71,9 +71,12 @@ class Uphold {
...
+      bool created,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.h]: const

**Pull request merged**